### DISCUSSION TITLE
Exporting PointRend to ONNX

### DISCUSSION THREAD

**Discussion by rvdinter** [facebookresearch/detectron2#3128] [Q&A]
Hi, I am trying to export my PointRend model to be used in ML.NET. To my understanding, this can only be done through exporting to ONNX format. Now, I see on [this](https://detectron2.readthedocs.io/en/latest/tutorials/deployment.html) documentation website, that this is not supported. Is there any way I can still export the model to ONNX format, or use the tracing model in ML.NET?

**Comment by thiagocrepaldi** [comment]
I am working on adding support to ONNX export (with and without caffe2) https://github.com/facebookresearch/detectron2/pull/4120 and https://github.com/facebookresearch/detectron2/pull/4153/

Hopefully it gets accepted soon

For the PointRend in specific, we will need to add grid_sampler operator to PyTorch, which will be done after these 2 PRs

**Reply by thiagocrepaldi** [reply]
https://github.com/facebookresearch/detectron2/pull/4413 is a unit test for PointRend. ONNX Export is already supported by main branch
